File: src/main/java/com/hoi4utils/clauzewitz/code/effect/EffectSchema.java
package main.java.com.hoi4utils.clauzewitz.code.effect;

import main.java.com.hoi4utils.clauzewitz.code.scope.ScopeType;

import java.util.EnumSet;
import java.util.List;

public record EffectSchema(List<String> pdxIdentifiers, EnumSet<ScopeType> supportedScopes,
                           EnumSet<ScopeType> supportedTargets) {

    public EffectSchema(String pdxIdentifier, EnumSet<ScopeType> supportedScopes) {
        this(List.of(pdxIdentifier), supportedScopes, null);
    }
}

File: src/main/scala/com/hoi4utils/databases/effect/EffectSchema.scala
package com.hoi4utils.databases.effect

import com.hoi4utils.hoi4mod.scope.ScopeType


final class EffectSchema(pdxIdentifiers: List[String], supportedScopes: Set[ScopeType], supportedTargets: Set[ScopeType]) {
  def this(pdxIdentifier: String, supportedScopes: Set[ScopeType]) = {
    this(List(pdxIdentifier), supportedScopes, Set.empty)
  }

  def this(pdxIdentifier: String, supportedScopes: Option[Set[ScopeType]], supportedTargets: Option[Set[ScopeType]]) = {
    this(List(pdxIdentifier), supportedScopes.get, supportedTargets.get)
  }
}
